packages/itmat-ui-react/src/components/lxd/lxd.instance.console.tsx

```diff
@@ -0,0 +1,177 @@
+                height: height,
+                type: 'vga'
+            })
+        });
+
+        if (!res) {
+            return;
+        }
+        const result = await res.json();
+
+        const baseUrl = new URL(window.location.href);
+        baseUrl.protocol = window.location.protocol === 'https:' ? 'wss:' : 'ws:';
+        baseUrl.pathname = '/rtc';
+
+        const dataUrl = `${baseUrl.href}?t=d&o=${result.operationId}&s=${result.operationSecrets['0']}`;
+        const controlUrl = `${baseUrl.href}?t=c&o=${result.operationId}&s=${result.operationSecrets.control}`;
+
+        const control = new WebSocket(controlUrl);
+
+        try {
+            window.spice_connection = new SpiceMainConn({
+                uri: dataUrl,
+                screen_id: 'spice-screen',
+                onerror: (e) => {
+                    console.error('Error in spice connection', e);
+                },
+                onsuccess: () => {
+                    handleResize();
+                },
+                onagent: handleResize
+            });
+
+            // // Clipboard from client to server
+            // document.addEventListener('copy', (e) => {
+            //     const clipboardData = e.clipboardData;
+            //     if (clipboardData) {
+            //         const text = clipboardData.getData('text/plain');
+            //         // If window.spice_connection is potentially undefined, check if it's defined before using it
+            //         window.spice_connection?.agent.clipboard_grab(text, SpiceMainConn.TYPE_TEXT);
+            //     }
+            //     e.preventDefault(); // Prevent the default copy action
+            // });
+
+            // // Clipboard from server to client
+            // window.spice_connection.add_channel_event_listener('spice_channel', 'agent', (agent, type) => {
+            //     if (type === SpiceMainConn.TYPE_CLIPBOARD) {
+            //         agent.clipboard_request(SpiceMainConn.TYPE_TEXT);
+            //     } else if (type === SpiceMainConn.TYPE_CLIPBOARD_SELECTION) {
+            //         // Handle clipboard selection if needed
+            //     }
+            // });
+
+            // window.spice_connection.agent.addEventListener('message', (e) => {
+            //     if (e.type === 'clipboard') {
+            //         const text = e.data; // The text from the Spice server's clipboard
+            //         navigator.clipboard.writeText(text).then(() => {
+            //             console.log('Clipboard text copied to client.');
+            //         }).catch((err) => {
+            //             console.error('Failed to copy text to clipboard:', err);
+            //         });
+            //     }
+            // });
+        } catch (e) {
+            console.error('Spice create error', e);
+        }
+
+
+        return control;
+    }, [instanceName]);
+
+
+    useEffect(() => {
+
+        // TODO !!!
+        // Should be removed and use proper async traps
```

Comment:
This from my original "demo" has not been addressed. It must be done before we release anything